### PR TITLE
Get Optics - DOM 

### PR DESCRIPTION
**Pull request by susanhooks**
Request to add method to EOS to address #1581 .

### CODE REVIEW

**Comment by jvanderaa** [comment]
@ktbyers thoughts on how this looks? Also, able to kick off the tests? I think there may have been some challenges with the testing that some quick guidance can be given.

File: napalm/eos/eos.py [hunk 1]
```diff
@@ -1877,6 +1877,67 @@ def get_optics(self):
 
         return optics_detail
 
+    def get_optics_dom(self):
+        """Implementation of "show interfaces transciever dom", allowing devices to
```
Comment (chadell) [inline napalm/eos/eos.py:1881]:
```suggestion
        """Implementation of "show interfaces transceiver dom", allowing devices to
```

**Comment by bewing** [comment]
Still not crazy about the idea of an entirely new getter -- might want to consider updating the base `get_optics` somehow so that both the old output (channels on subsumed interfaces) + the new output (multiple channels on the main interface) are supported.

Also, this throws an exception on 4.22.5M and 4.23.8M, as ports without optics inserted in them do not have a `parameters` dictionary in them:

```python
In [19]: resp[0]["interfaces"]
Out[19]:
{'Ethernet8': {'displayName': 'Ethernet8'},
 'Ethernet9': {'displayName': 'Ethernet9'},
 'Ethernet2': {'displayName': 'Ethernet2'},
 'Ethernet3': {'displayName': 'Ethernet3'},
 'Ethernet1': {'displayName': 'Ethernet1'},
 'Ethernet6': {'displayName': 'Ethernet6'},
 'Ethernet7': {'displayName': 'Ethernet7'},
 'Ethernet4': {'displayName': 'Ethernet4'},
 'Ethernet5': {'displayName': 'Ethernet5'},
 'Ethernet52/1': {'updateTime': 1649799051.5188503,
  'displayName': 'Ethernet52/1',
  'parameters': {'txPower': {'channels': {'1': -0.22962664775318586,
```

**Comment by susanhooks** [comment]
Closing PR, have decided to address with a custom method to extend the base.

**Pull request closed without merging**